### ISSUE TITLE
Purging for images (binary files) makes no effect

### ISSUE THREAD

**Issue by jcubic** [jsdelivr/jsdelivr#18520]
### Description

I've added png images (avatars) to my static GitHub repo to be able to use them on the GitHub sponsors page.

I'm not able to purge the URL in any way. I used `@master` `@latest` also this page:

https://www.jsdelivr.com/tools/purge

It has no effect. I can only see the updated file if I use commit hash.

### Affected jsDelivr links

https://cdn.jsdelivr.net/gh/jcubic/static/assets/avatar.png
https://cdn.jsdelivr.net/gh/jcubic/static/assets/avatar-small.png

### Response headers

```
< HTTP/2 200 
< date: Thu, 17 Aug 2023 20:12:54 GMT
< content-type: image/png
< content-length: 346820
< access-control-allow-origin: *
< access-control-expose-headers: *
< timing-allow-origin: *
< cache-control: public, max-age=604800, s-maxage=43200
< cross-origin-resource-policy: cross-origin
< x-content-type-options: nosniff
< strict-transport-security: max-age=31536000; includeSubDomains; preload
< x-jsd-version: master
< x-jsd-version-type: branch
< etag: W/"54ac4-8xXRU+Rpf3FTC8ZXwmTNPn2kqsA"
< x-served-by: cache-fra-etou8220103-FRA, cache-bma1624-BMA
< x-cache: MISS, MISS
< vary: Accept-Encoding
< alt-svc: h3=":443"; ma=86400
< cf-cache-status: HIT
< age: 431
< accept-ranges: bytes
< report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v3?s=BnX5Drb3JdYvQrMHLYmJ2rZkQmU5DqQ4GBnJBX%2BOceZVRKCnkGNzyNcGoFMuYKXx%2FeiTEvXvG4UTZCc6Pg%2B7C8RewYZIeUn4bpP9spEvYnU%2B%2FznQHCX8q4CtenWSXMIeYX0%3D"}],"group":"cf-nel","max_age":604800}
< nel: {"success_fraction":0.01,"report_to":"cf-nel","max_age":604800}
< server: cloudflare
< cf-ray: 7f8493168b3f5012-WAW
```

### Information

- Device OS: [e.g. iOS, Windows, MacOS] Browser
- Browser & Version: [e.g. Chrome 103, Safari 8, Firefox 108] Chrome 115
- VPN provider (if you use one)
- Your location (country/region) Poland

### Requisites

- [X] I performed a [cursory search of the issue tracker](https://github.com/jsdelivr/jsdelivr/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] I checked the [documentation](https://github.com/jsdelivr/jsdelivr) to understand that the issue I am reporting is not normal behavior.
- [X] I understand that not filling out this template correctly will lead to the issue being closed.

**Comment by MartinKolarik** [comment]
The API is meant to work with semver tags, such as `v1.2.3`, which can then be referenced as `@1` or `@latest` in the CDN links. We don't currently support purging for branches (it may work sometimes, but not reliably).

**Comment by jcubic** [comment]
It is because it's the binary file? I often can purge js files without any issues. I often add a branch name and get the latest version that way. But with an image, nothing works.

**Comment by MartinKolarik** [comment]
No, that is not related. Since branch files can't be easily "resolved" to a fixed version, they are cached differently and purging doesn't take that into account, currently.
